src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -1447,6 +1451,47 @@ public DateTimeFormatterBuilder appendLiteral(String literal) {
         return this;
     }
 
+    /**
+     * Appends the day period text to the formatter.
+     * <p>
+     * This appends an instruction to format/parse the textual name of the day period
+     * to the builder. Day periods are defined in LDML's
+     * <a href="https://unicode.org/reports/tr35/tr35-dates.html#dayPeriods">"day periods"
+     * </a> element.
+     * <p>
+     * During formatting, the day period is obtained from {@code HOUR_OF_DAY}, and
+     * optionally {@code MINUTE_OF_HOUR} if exist. It will be mapped to a day period
+     * type defined in LDML, such as "morning1" and then it will be translated into
+     * text. Mapping to a day period type and its translation both depend on the
+     * locale in the formatter.
+     * <p>
+     * During parsing, the text will be parsed into a day period type first. If
+     * {@code HOUR_OF_DAY}, and optionally {@code MINUTE_OF_HOUR} exists,
+     * those values are validated with the parsed day period. If they are within the
+     * parsed day period, those values remain intact, otherwise {@code DateTimeException}
+     * is thrown. If {@code HOUR_OF_AMPM} exists and validated with the day period,
+     * {@code HOUR_OF_DAY} is derived from the parsed day period. If neither
+     * {@code HOUR_OF_DAY} nor {@code HOUR_OF_AMPM} exists, {@code HOUR_OF_DAY} and
+     * {@code MINUTE_OF_HOUR} are resolved to the mid-point of the day period.
+     * For example, if the parsed day period type is "night1" and the period defined
+     * for it in the formatter locale is from 21:00 to 06:00, then {@code HOUR_OF_DAY}
+     * is set to '1' and {@code MINUTE_OF_HOUR} set to '30'. If any conflict occurs in
+     * {@link ResolverStyle#LENIENT LENIENT} mode, no exception is thrown and the day
+     * period is ignored.
+     *
+     * @param style the text style to use, not null
+     * @return this, for chaining, not null
+     * @since 16
+     */
+    public DateTimeFormatterBuilder appendDayPeriodText(TextStyle style) {
+        Objects.requireNonNull(style, "style");
+        if (style != TextStyle.FULL && style != TextStyle.SHORT && style != TextStyle.NARROW) {
+            throw new IllegalArgumentException("Style must be either full, short, or narrow");
```

Comment:
Nothing in the Javadoc describes this behaviour. It would make more sense to map FULL_STANDALONE to FULL and so on and not throw an exception.

Reply:
Fixed.